lib/matobj2.gd

```diff
@@ -348,25 +348,49 @@ DeclareOperation( "AddRowVector",
 
 
 
-# TODO: rename MultRowVector to MultVector; but keep in mind that
-# historically there already was MultRowVector, so be careful to not break that
-DeclareOperation( "MultRowVector",
-  [ IsVectorObj and IsMutable, IsObject ] );
-
-#
-# Also, make it explicit from which side we multiply
-# DeclareOperation( "MultRowVectorFromLeft",
-#   [ IsVectorObj and IsMutable, IsObject ] );
-# DeclareOperation( "MultRowVectorFromRight",
-#   [ IsVectorObj and IsMutable, IsObject ] );
-#DeclareSynonym( "MultRowVector", MultRowVectorFromRight );
-
-# do we really need the following? for what? is any code using this right now?
-# ( a, pa, b, pb, s ) ->  a{pa} := b{pb} * s;
-DeclareOperation( "MultRowVector",
-  [ IsVectorObj and IsMutable, IsList, IsVectorObj, IsList, IsObject ] );
-
-# maybe have this:   vec := vec{[from..to]} * scal ?? cvec has it
+# TODO: rename MultVector to MultVector; but keep in mind that
+# historically there already was MultVector, so be careful to not break that
+#############################################################################
+##
+#O  MultVector( <vec>, <mul>[, <from>, <to>] )
+#O  MultVectorLeft( <vec>, <mul>[, <from>, <to>] )
+#O  MultVectorRight( <vec>, <mul>[, <from>, <to>] )
+##
+##  <#GAPDoc Label="MatObj_MultVector">
+##  <#GAPDoc Label="MatObj_MultVectorLeft">
+##  <#GAPDoc Label="MatObj_MultVectorRight">
+##  <ManSection>
+##  <Oper Name="MultVectorLeft" Arg='vec, mul[, from, to]'/>
+##  <Oper Name="MultVectorRight" Arg='vec, mul[, from, to]'/>
+##  <Returns>nothing</Returns>
+##
+##  <Description>
+##  Note that <C>MultVector</C> is only a Synonym for
+##  <C>MultVectorLeft</C>.
+##  </P>
+##  These operations multiply <A>vec</A> with <A>mul</A> in-place and write
+##  the result to <A>vec</A>.
+##  <C>MultVectorLeft</C> multiplies <A>mul</A> to <A>vec</A> from the
+##  left. <C>MultVectorRight</C> does so from the right.
+##  </P>
+##  The optional parameters <A>from</A> and <A>to</A> can be used to restrict
+##  the effect of the operation to the entries
+##  <C><A>vec</A>{[<A>from</A>..<A>to</A>]}.
```

Comment:
Hmm, this suggest that this syntax is permissible for vector objs, but in general it isn't. Please rewrite in plain words, or perhaps as :... to the entries vec[from], ..., vec[to]."

Perhaps also add something like: "if from > to, then ..."  and describe what we expect will happen (personally, I'd suggest nothing should happen, just like with `vec{[from..to}]`, which is an empty list in that case; but of course one could also raise an error. In any case, we should specify it.

Reply:
@fingolfin Okay, then I'll have to change `AddRowVector` and its documentation accordingly.

I see a problem with `MultRowVector` and `AddRowVector` behaving differently in the cvec package though. The documentation states (using `fr` instead of `from`)

> If either `fr` or `to` is 0 it defaults to 1 and Length(v) respectively.

In this case calling `MultRowVector( vec, mul, 2, 1 )` would do nothing, but according to the currently documented behaviour `MultRowVector( vec, mul, 2, 0 )` still has to perform the multiplication on all entries (except for possibly the first one).

Should I also do a PR for the cvec package then or leave it as it is?